### ISSUE TITLE
Add repository support for Ubuntu 22.04 Jammy Jellyfish

### ISSUE THREAD

**Issue by julian-alarcon** [OpenVPN/openvpn3-linux#113]
There is currently no repository file for Ubuntu 22.04.
Using the official installation documentation located [here](https://openvpn.net/cloud-docs/openvpn-3-client-for-linux/) I got the error 404 Not Found

```
wget -O /etc/apt/sources.list.d/openvpn3.list https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-$DISTRO.list                                                                                                                                                          
--2022-03-10 16:26:51--  https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-.list                                                                                                                                                                                                                                  
Resolving swupdate.openvpn.net (swupdate.openvpn.net)... 104.18.109.96, 104.18.110.96                                                                                                                                                                                                                                          
Connecting to swupdate.openvpn.net (swupdate.openvpn.net)|104.18.109.96|:443... connected.                                                                                                                                                                                                                                     
HTTP request sent, awaiting response... 404 Not Found                                                                                                                                                                                                                                                                          
2022-03-10 16:26:52 ERROR 404: Not Found.             
```

**Comment by julian-alarcon** [comment]
I got an error trying to install the openvpn3 client using the Ubuntu 21.10 repository
```
$ sudo apt install openvpn3
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 openvpn3 : Depends: libjsoncpp24 (>= 1.9.4) but it is not installable
            Depends: libtinyxml2-8 (>= 8.0.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

Those missing dependencies have newer versions in Ubuntu 22.04.

**Comment by dsommers** [comment]
We will do a proper build for newer Ubuntu releases once they are released. We do not provide any builds on Ubuntu pre-releases.

It is required a bit more than just adding a new repository.  As you already spotted, using the 21.10 build will not work, due to dependencies has been updated as well.  When dependencies are upgraded to a new base release, OpenVPN 3 Linux needs to be recompiled with those dependencies.

**Comment by schwabe** [comment]
Compile openvpn3-linux from source if you need it now on Ubuntu 22.04

**Comment by julian-alarcon** [comment]
Soo.. Ubuntu 22.04 is now a public and stable release https://ubuntu.com/blog/ubuntu-22-04-lts-released also, it's marked as a LTS (Long Time Support).

**Comment by Sinnv2710** [comment]
> Soo.. Ubuntu 22.04 is now a public and stable release https://ubuntu.com/blog/ubuntu-22-04-lts-released also, it's marked as a LTS (Long Time Support).

Hopefully, Openvpn3-Linux will have release on 22.04 ASAP

**Comment by dsommers** [comment]
@julian-alarcon @Sinnv2710 If you need it urgently, I recommend you to pull down the source code from git (`git clone https://github.com/OpenVPN/openvpn3-linux.git`), install needed dependencies as described in the [README.md](https://github.com/OpenVPN/openvpn3-linux/blob/master/README.md#debianubuntu) file and then follow the [build instructions](https://github.com/OpenVPN/openvpn3-linux/blob/master/README.md#preparations-building-from-git).

If you find any issues on Ubuntu 22.04 LTS, please let us know so we can fix those as soon as possible; report issues in new issue tickets or related tickets (**not** this ticket - as this is about pre-built packages for 22.04 LTS, not issues running/building on that distribution).

**Comment by julian-alarcon** [comment]
Don't worry about me @dsommers :) . I have it working since a long time.
I created this ticket thinking on the people that is struggling with this from my close circle.

**Comment by Sinnv2710** [comment]
Hi @julian-alarcon, how you can install openvpn3 on ubuntu 22? I tried build from the source code but not working

**Comment by julian-alarcon** [comment]
You can find how I did it here https://github.com/OpenVPN/openvpn3-linux/issues/113#issuecomment-1064527239. Not the recommended way, but this way keep using .deb files from a recognized source. Just wait for the new version with support of Ubuntu 22.04.

**Comment by Sinnv2710** [comment]
I can install it follow by your comment.
thank you very much.

**Comment by AP-TGR** [comment]
Is there any estimated time when we will get the support for Ubuntu 22.04?

**Comment by bagpussnz** [comment]
waiting for openvpn to get back to me for an eta of 22.04 repo

**Comment by cron2** [comment]
Hi,

On Mon, Apr 25, 2022 at 06:45:39PM -0700, bagpussnz wrote:
> waiting for openvpn to get back to me for an eta of 22.04 repo

I'll escalate this to the service manager to have them refund your
maintenance fees if this cannot be resolved in a timely manner.

gert

-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by bagpussnz** [comment]
I've already tried compiling from github... had dbus issues. No problems...
I'll wait. Can use openvpn2
Regards,
Ian Collins

**Comment by bagpussnz** [comment]
Hi - It installed and connected fine.
I'll use it for connecting remotely to work and see how it goes.

Many Thanks,
Ian Collins

**Comment by tajnosc** [comment]
@dsommers there is a problem with that repository file

$ sudo apt update
Hit:1 http://mt.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://security.ubuntu.com/ubuntu jammy-security InRelease                                                                           
Hit:3 http://mt.archive.ubuntu.com/ubuntu jammy-updates InRelease                                                                          
Hit:4 https://swupdate.openvpn.net/community/openvpn3/repos jammy InRelease                                                                
Hit:5 http://mt.archive.ubuntu.com/ubuntu jammy-backports InRelease                                                                        

Reading package lists... Done                 
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://swupdate.openvpn.net/community/openvpn3/repos jammy InRelease' doesn't support architecture 'i386'

**Comment by dsommers** [comment]
@tajnosc Have you really installed a Ubuntu 22.04 32bit Intel (i386) release on a system in 2022?

OpenVPN 3 Linux does currently not target 32 bit systems at all.  Support for 32 bit OSes is _not_ something we have high up on our todo lists.

**Comment by schwabe** [comment]
@tajnosc 32 bit i386 is not a supported distribution from our side. If you ended up with multiarch try forcing amd64 only by using 
```
deb  [arch=amd64] https://swupdate.openvpn.net/community/openvpn3/repos jammy main
```

**Comment by tajnosc** [comment]
Oh, yes sorry. Forcing amd64 removed that warning.

**Comment by julian-alarcon** [comment]
I confirm! using this repo:
```
deb [arch=amd64] https://swupdate.openvpn.net/community/openvpn3/repos jammy main
```

Works flawlessly! I was able to remove the external .deb packages installed before and now it's using the normal Jammy packages!

**Comment by wa1one** [comment]
The repository 'https://swupdate.openvpn.net/community/openvpn3/repos jammy InRelease' is not signed.

**Comment by MahadevMG** [comment]
I am able to install open vpn 3 on ubuntu 22.04 and i can use it without any issue.
This may not be a proper way but this worked for me.

Step 1:  Checked is there any  `openvpn-repo-pkg-keyring.gpg` file under `/etc/apt/trusted.gpg.d/ ` ( I removed this file)
Step 2: Checked is there any `openvpn3.list` under /etc/apt/sources.list.d/ ( i removed this too - i had these files because i tried couple of ways to install)

Step 3: `sudo apt update`
Step 4:  `sudo apt install apt-transport-https`
Step 5: `sudo wget https://swupdate.openvpn.net/repos/openvpn-repo-pkg-key.pub`
Step 6: `sudo apt-key add openvpn-repo-pkg-key.pub`
Step 7: `sudo wget -O /etc/apt/sources.list.d/openvpn3.list https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-$DISTRO.list`

I changed $DISTRO to jammy 
ie : `sudo wget -O /etc/apt/sources.list.d/openvpn3.list https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-jammy.list`

Step 8: `sudo apt update`
Step 9: `sudo apt install openvpn3`

**Comment by bagpussnz** [comment]
Hi - sorry for the delay... will try it tonight (NZ time) - and thank you.

Regards,
Ian Collins

**Comment by wa1one** [comment]
The repository 'https://swupdate.openvpn.net/community/openvpn3/repos jammy InRelease' is not signed.

**Comment by SquirtleSquadLeader** [comment]
> The repository 'https://swupdate.openvpn.net/community/openvpn3/repos jammy InRelease' is not signed.

@wa1one I am new to VPNs.  Does it not being signed imply that it is not safe?  Sorry for my ignorance... Just trying to make an informed decision if I should proceed with that repo.  Thank you!

**Comment by dsommers** [comment]
@SquirtleSquadLeader It should be signed now.

**Comment by ShreeBaburajan** [comment]
Followed all steps in 

@MahadevMG posted code:

I am able to install openvpn3 on ubuntu 22.04 and i can use it without any issue.
This may not be a proper way but this worked for me.

Step 1: Checked is there any openvpn-repo-pkg-keyring.gpg file under /etc/apt/trusted.gpg.d/ ( I removed this file)
Step 2: Checked is there any openvpn3.list under /etc/apt/sources.list.d/ ( i removed this too - i had these files because i tried couple of ways to install)

Step 3: sudo apt update
Step 4: sudo apt install apt-transport-https
Step 5: sudo wget https://swupdate.openvpn.net/repos/openvpn-repo-pkg-key.pub
Step 6: sudo apt-key add openvpn-repo-pkg-key.pub
Step 7: sudo wget -O /etc/apt/sources.list.d/openvpn3.list https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-$DISTRO.list

I changed $DISTRO to jammy
ie : sudo wget -O /etc/apt/sources.list.d/openvpn3.list https://swupdate.openvpn.net/community/openvpn3/repos/openvpn3-jammy.list

Step 8: sudo apt update


But after step 8 do the oresle I kept getting errors
In Softwares % Updates, under the "Ubuntu Software', check all boxes in Downloadable from Internet, then follow step 9

Step 9: sudo apt install openvpn3

Thanks @MahadevMG

**Comment by riyadh9292** [comment]
@ShreeBaburajan  thnx... That saves me.
